docs/topics/db/sql.txt

```diff
@@ -4,8 +4,33 @@ Performing raw SQL queries
 
 .. currentmodule:: django.db.models
 
-When the :doc:`model query APIs </topics/db/queries>` don't go far enough, you
-can fall back to writing raw SQL. Django gives you two ways of performing raw
+.. Important:: 
+    
+    As of Django 2.1, there should be no reason to have to drop into raw SQL to 
+    perform complex queries. 
+
+    * Calculated fields can be created using :doc:`aggregations </topics/db/aggregation>` and `annotations`_.
```

Comment:
I'm not sure this list is needed -- although you might be able to work some links throughout the doc to point out the non-raw alternatives (for example, under "Adding annotations" recommend query expressions).

Reply:
While that might be useful to work into the document, what this document really needs right at the top is an updated version of "No, really, you shouldn't need to use this." with references to the major alternatives. 

While I don't have the full core-dev-level of knowledge advocate for `raw()` or `extra()` to be removed entirely, the _least_ we can do is update the landing page to really talk people out of using it.

Reply:
I'm more than happy to accept wording suggestions to this PR, but *something* needs to be added to the top of this page. The ORM has changed too much for the current summary/warnings to be accurate.

Reply:
I think we all agree that is document is quite outdated and would greatly benefit from a warning that `raw()` is almost certainly not required at this point.

[A recent post on Reddit](https://old.reddit.com/r/django/comments/9u3mt5/i_made_a_cheat_sheet_for_django_orm_optimization/) that was comparing _raw_ SQL queries to how they can be constructed using the ORM was very well received. I wonder if we could use a similar approach here instead of a list to convince the reader that `raw()` is not required in these cases.